templates/run_vdbench.yml

```diff
@@ -0,0 +1,55 @@
+# Copyright (C) Microsoft Corporation. All rights reserved.
+# https://aka.ms/yaml
+parameters:
+- name: cmd_terraform
+  type: string
+  values:
+  - "apply"
+  - "destroy"
+- name: vdbench_config
+  type: string
+  default: "ondisk"
+  values:
+    - "ondisk"
+    - "inmem"
+
+steps:
+- script: |
+    cd src/terraform/examples/vfxt/${RUN_TERRAFORM_FILE}
+    CONTROLLER_ADDRESS=$(terraform output -json | jq -r ".controller_address.value")
+    USERNAME=$(terraform output -json | jq -r ".controller_username.value")
+    VSERVER_IP_ADDRESSES=$(terraform output -json | jq -c ".mount_addresses.value")
+    NFS_EXPORT_PATH=$(terraform output -json | jq -r ".mount_namespace_path.value")
+    ssh_key_data="$(cat ~/.ssh/id_rsa.pub) azureuser@linuxvm"
+    echo "##vso[task.setvariable variable=secret.SSH_KEY_DATA;issecret=true]${ssh_key_data}"
+    cd ../vdbench
+
+    terraform init
+    TF_LOG=INFO terraform ${{ parameters.cmd_terraform }} -auto-approve -var="location=${REGION}" \
+      -var="vmss_resource_group_name=${RESOURCE_GROUP_NAME}-vmss" \
+      -var="controller_address=${CONTROLLER_ADDRESS}" \
+      -var="controller_username=${USERNAME}" \
+      -var="ssh_key_data=${ssh_key_data}" \
+      -var="vserver_ip_addresses=${VSERVER_IP_ADDRESSES}" \
+      -var="nfs_export_path=${NFS_EXPORT_PATH}" \
+      -var="vdbench_url=${VDBENCH_URL}" \
+      -var="vnet_resource_group=${RESOURCE_GROUP_NAME}-network"
+
+    if [ "apply" = "${{ parameters.cmd_terraform }}" ]; then
+      IP_ADDRESS_COMMAND=$(terraform output -json | jq -r ".vmss_addresses_command.value") 
+      IP_ADDRESS=$(${IP_ADDRESS_COMMAND} | jq -r '.[0]')
+      
+      ssh -o "StrictHostKeyChecking=no" -oProxyCommand="ssh -oStrictHostKeyChecking=no -W %h:%p  ${USERNAME}@${CONTROLLER_ADDRESS}" ${USERNAME}@${IP_ADDRESS} ls
+      echo ">> SCP CMD"
+      scp -o "ProxyJump ${USERNAME}@${CONTROLLER_ADDRESS}" ~/.ssh/id_rsa ${USERNAME}@${IP_ADDRESS}:.ssh/id_rsa
+      
+      echo ">> COPY_IDARSA.SH"
+      ssh -J ${USERNAME}@${CONTROLLER_ADDRESS} ${USERNAME}@${IP_ADDRESS} /home/azureuser/copy_idrsa.sh
+      
+      echo ">> RUN VDBENCH"
+      ssh -J ${USERNAME}@${CONTROLLER_ADDRESS} ${USERNAME}@${IP_ADDRESS} "tmux new-session -d \"/home/${USERNAME}/run_vdbench.sh ${{ parameters.vdbench_config }}.conf uniquestring1 >> /home/${USERNAME}/vdb.log 2>&1\""
```

Comment:
If we query the whole expected time, it will not move on to scale up / scale down and Anthony wants to run those while load is running. Unsure what I should do?

Reply:
@anhowe Let me know what you think?

Reply:
Let's push this to future, here's the issue https://github.com/Azure/Avere/issues/1066.